src/CodeGen_LLVM.cpp

```diff
@@ -499,19 +513,78 @@ MangledNames get_mangled_names(const LoweredFunc &f, const Target &target) {
 
 }  // namespace
 
-std::unique_ptr<llvm::Module> CodeGen_LLVM::compile(const Module &input) {
+// Make a wrapper to call the function with an array of pointer
+// args. This is easier for the JIT to call than a function with an
+// unknown (at compile time) argument list.
+llvm::Function *CodeGen_LLVM::add_trampoline_wrapper(llvm::Function *fn,
```

Comment:
How similar is this to the argv entrypoint?

Reply:
Similar-ish. It should really be reconciled to avoid somewhat-duplicated code.

Reply:
The comment should explain why the argv version doesn't serve your purposes. It seems like the only difference is that the scalars are done indirectly. Is that because there are scalar types wider than the pointer type in wasm?

Reply:
Oh, I think I misunderstood what this is for. I thought it was another wrapper for the pipeline, but it's a wrapper for extern functions?

Reply:
That's correct. (It's basically directly lifted from @zvookin's earlier work on a JS backend.)